pkg/monitor/resourcewatch/controller/configmonitor/crd_controller.go

```diff
@@ -0,0 +1,163 @@
+package configmonitor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	apiextensionsv1beta1lister "k8s.io/apiextensions-apiserver/pkg/client/listers/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var (
+	defaultResyncDuration = 1 * time.Minute
+)
+
+type ConfigObserverController struct {
+	crdLister          apiextensionsv1beta1lister.CustomResourceDefinitionLister
+	crdInformer        cache.SharedIndexInformer
+	dynamicClient      dynamic.Interface
+	dynamicInformers   []*dynamicConfigInformer
+	cachedDiscovery    discovery.CachedDiscoveryInterface
+	monitoredResources []schema.GroupVersion
+	storageHandler     cache.ResourceEventHandler
+}
+
+func NewConfigObserverController(
+	dynamicClient dynamic.Interface,
+	crdInformer cache.SharedIndexInformer,
+	discoveryClient *discovery.DiscoveryClient,
+	configStorage cache.ResourceEventHandler,
+	monitoredResources []schema.GroupVersion,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &ConfigObserverController{
+		dynamicClient:      dynamicClient,
+		crdInformer:        crdInformer,
+		storageHandler:     configStorage,
+		monitoredResources: monitoredResources,
+		cachedDiscovery:    memory.NewMemCacheClient(discoveryClient),
+	}
+	c.crdLister = apiextensionsv1beta1lister.NewCustomResourceDefinitionLister(c.crdInformer.GetIndexer())
+
+	return factory.New().WithInformers(c.crdInformer).ResyncEvery(defaultResyncDuration).WithSync(c.sync).ToController("ConfigObserverController", recorder.WithComponentSuffix("config-observer-controller"))
+}
+
+// currentResourceKinds returns list of group version configKind for OpenShift configuration types.
+func (c *ConfigObserverController) currentResourceKinds() ([]schema.GroupVersionKind, error) {
+	observedCrds, err := c.crdLister.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	var (
+		currentConfigResources []schema.GroupVersionKind
+		currentKinds           = sets.NewString()
+	)
+	for _, crd := range observedCrds {
+		for _, gv := range c.monitoredResources {
+			if !strings.HasSuffix(crd.GetName(), "."+gv.Group) {
+				continue
+			}
+			for _, version := range crd.Spec.Versions {
+				if !version.Served {
+					continue
+				}
+				gvk := schema.GroupVersionKind{
+					Group:   gv.Group,
+					Version: gv.Version,
+					Kind:    crd.Spec.Names.Kind,
+				}
+				if currentKinds.Has(gvk.Kind) {
+					continue
+				}
+				currentKinds.Insert(gvk.Kind)
+				currentConfigResources = append(currentConfigResources, gvk)
```

Comment:
nit: you could just condense these into a `if !currentKinds.Has(gvk.Kind)` block and drop the `continue`